### PR TITLE
RAV-1410 : Handling TO_NUMBER for snowflake

### PR DESCRIPTION
**Pull request by vaibhavjain-dm**
RAV-1410 : Handling TO_NUMBER for snowflake

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/SqlKind.java [hunk 1]
```diff
@@ -207,6 +207,11 @@
    */
   INTERSECT,
 
+  /**
+   * Conv
+   */
+  CONV,
```
Comment (soma-mondal) [inline core/src/main/java/org/apache/calcite/sql/SqlKind.java]: Please use other function as SqlKind

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 2]
```diff
@@ -478,6 +478,16 @@ private SqlLibraryOperators() {
           OperandTypes.family(SqlTypeFamily.NULL)),
           SqlFunctionCategory.STRING);
 
+  @LibraryOperator(libraries = {STANDARD})
```
Comment (soma-mondal) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java]: Check if Conv is standard function

File: core/src/main/java/org/apache/calcite/util/ToNumberUtils.java [hunk 3]
```diff
@@ -31,134 +31,253 @@
...
+    }
+  }
+
+  public static void unparseToNumberSnowFlake(SqlWriter writer, SqlCall call,
+                                              int leftPrec, int rightPrec) {
+    switch (call.getOperandList().size()) {
+    case 1:
+    case 3:
+      SqlNode[] extractNodeOperands;
+      if (isOperandLiteral(call)) {
```
Comment (soma-mondal) [inline core/src/main/java/org/apache/calcite/util/ToNumberUtils.java]: Remove this if check

File: core/src/main/java/org/apache/calcite/util/ToNumberUtils.java [hunk 4]
```diff
@@ -31,134 +31,253 @@
...
-          ? SqlTypeName.FLOAT : SqlTypeName.INTEGER;
-      handleCasting(writer, call, leftPrec, rightPrec, sqlTypeName);
+    SqlCall extractCallCast = new SqlBasicCall(SqlStdOperatorTable.CAST,
+            extractNodeOperands, SqlParserPos.ZERO);
 
-      break;
-    case 2:
-      if (handleIfOperandIsNull(writer, call, leftPrec, rightPrec)) {
-        return;
+    SqlStdOperatorTable.CAST.unparse(writer, extractCallCast, leftPrec, rightPrec);
```
Comment (soma-mondal) [inline core/src/main/java/org/apache/calcite/util/ToNumberUtils.java]: We should use dialect.unparseCall

File: core/src/main/java/org/apache/calcite/sql/SqlKind.java (hunk 1 above)
Comment (vaibhavjain-dm) [reply core/src/main/java/org/apache/calcite/sql/SqlKind.java]: Done.

File: core/src/main/java/org/apache/calcite/util/ToNumberUtils.java (hunk 3 above)
Comment (vaibhavjain-dm) [reply core/src/main/java/org/apache/calcite/util/ToNumberUtils.java]: Done

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java (hunk 2 above)
Comment (vaibhavjain-dm) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java]: Done

File: core/src/main/java/org/apache/calcite/util/ToNumberUtils.java (hunk 4 above)
Comment (vaibhavjain-dm) [reply core/src/main/java/org/apache/calcite/util/ToNumberUtils.java]: Done

**Pull request merged**